src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep.java

```diff
@@ -125,15 +127,25 @@
     }
 
     /** Similar to {@code MaskPasswordsOutputStream}. */
-    private static final class Filter extends ConsoleLogFilter implements Serializable {
+    static final class Filter extends ConsoleLogFilter implements Serializable {
 
         private static final long serialVersionUID = 1;
 
         private final Secret pattern;
+        
+        private static final Comparator<String> StringLengthComparator = new Comparator<String>() {
+			@Override
+			public int compare(String o1, String o2) {
+				return o1.length() - o2.length();
+			}
+		};
 
         Filter(Collection<String> secrets) {
             StringBuilder b = new StringBuilder();
-            for (String secret : secrets) {
+            List<String> hiddenSecrets = new ArrayList<String>(secrets);
+            Collections.sort(hiddenSecrets, StringLengthComparator);
```

Comment:
Or something to do with substrings? Should be done as a separate PR. Nothing to do with the rest of this PR.